docs/components/DocumentationSidebar.tsx

```diff
@@ -22,12 +21,91 @@ const STYLES_SECTION_CATEGORY = css`
   margin-bottom: 24px;
 `;
 
-function shouldSkipCategory(info: NavigationRoute) {
-  if (info.name === 'Feature Preview') {
-    return true;
+type SidebarProps = {
+  url: Url;
+  asPath: string;
+  isVersionSelectorHidden: boolean;
+  routes: NavigationRoute[];
+  version: string;
+  onSetVersion: (value: string) => void;
+};
+
+type SidebarNodeProps = Pick<SidebarProps, 'url' | 'asPath'> & {
+  route: NavigationRoute;
+  parentRoute?: NavigationRoute;
+};
+
+// TODO(cedric): move navigation over to unist format and use type to select different "renderers"
+export default function DocumentationSidebar(props: SidebarProps) {
+  return (
+    <nav css={STYLES_SIDEBAR} data-sidebar>
+      {!props.isVersionSelectorHidden && (
+        <VersionSelector version={props.version} onSetVersion={props.onSetVersion} />
+      )}
+      {props.routes.map(section => (
+        <DocumentationSidebarSection
+          key={`section-${section.name}`}
+          route={section}
+          url={props.url}
+          asPath={props.asPath}
+        />
+      ))}
+    </nav>
+  );
+}
+
+function DocumentationSidebarSection(props: SidebarNodeProps) {
+  // If the section or group is hidden, we should not render it
+  if (props.route.hidden) {
+    return null;
   }
 
-  return false;
+  // If a group was passed instead of section, just render that instead
+  if (!props.route.children) {
+    return <DocumentationSidebarGroup {...props} />;
+  }
+
+  return (
+    <DocumentationSidebarCollapsible
+      key={`group-${props.route.name}`}
+      url={props.url}
+      info={props.route}
+      asPath={props.asPath}>
+      {props.route.children.map(group => (
+        <DocumentationSidebarGroup
+          {...props}
+          key={`group-${props.route.name}`}
+          route={group}
+          parentRoute={props.route}
+        />
+      ))}
+    </DocumentationSidebarCollapsible>
+  );
+}
+
+function DocumentationSidebarGroup(props: SidebarNodeProps) {
+  return (
+    <div css={STYLES_SECTION_CATEGORY}>
+      {!shouldSkipTitle(props.route, props.parentRoute) && (
```

Comment:
With the changes from PR #16094, we can also make sections dynamic to render either groups or pages directly. That's nice because we can just remove the `makeGroup([ ...makePage ])` wrap in navigation to achieve the same result here (without requiring to check if the title are the same).

Reply:
So instead of writing this structure on [feature preview's sidebar](https://docs.expo.dev/feature-preview/)
```
├─ section "Development Builds"
│  └─ group "Development Builds" // Title is hidden, exactly the same as section title
│     ├─ page "Introduction"
│     ├─ page "Getting Started"
│     ├─ page "Creating Development Builds"
│     └─ ... 
│
└─ section "EAS Update"
   └─ group "EAS Update" // Title is hidden, exactly the same as section title
      ├─ page "Introduction"
      ├─ page "Getting Started"
      ├─ page "Using GitHub Actions"
      └─ ... 
```

With PR #16094 we can iterate inside a **section** node, and either render a **group** OR **page** directly. Basically allowing us to write this instead, and not having to care about those titles:

```
├─ section "Development Builds"
│  ├─ page "Introduction"
│  ├─ page "Getting Started"
│  ├─ page "Creating Development Builds"
│  └─ ... 
│
└─ section "EAS Update"
   ├─ page "Introduction"
   ├─ page "Getting Started"
   ├─ page "Using GitHub Actions"
   └─ ... 
```